docs/backend/setup/database-migrations.md

```diff
@@ -98,56 +98,112 @@ When you are writing the code for your new migration, there are a few things you
-  UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
-}
+    Some things to note:
+
+    1. The SQL keywords don't need to be capitalized, but it can be nice to differentiate some of them from certain 
+       things like the field names or types.
+    2. We want to add columns for every column other than the `id`, `created_at`, and `updated_at` fields. This makes it
+       easier for future folks to understand what the columns were meant for.
+    3. You might see some migrations use `varchar` instead of `text`. It's the same thing in the end, see
+       [Postgresql Character Types](https://www.postgresqltutorial.com/postgresql-char-varchar-text/) for more info.
+    
+3. Now you can create a new file in `pkg/models/` named after your new model. So for this, we'll have a new file 
+    `pkg/models/cat.go` that will look something like this:
+
+    ```go
+    package models
+    
+    import (
+        "time"
+    
+        "github.com/gofrs/uuid"
+    
+        "github.com/transcom/mymove/pkg/unit"
+    )
+    
+    // Cat contains all the information relevant to a cat...
+    type Cat struct {
+        ID         uuid.UUID   `json:"id" db:"id"`
+        CreatedAt  time.Time   `json:"created_at" db:"created_at"`
+        UpdatedAt  time.Time   `json:"updated_at" db:"updated_at"`
+        Name       string      `json:"name" db:"name"`
+        Birthday   *time.Time  `json:"birthday" db:"birthday"`
+        GotchaDay  *time.Time  `json:"gotcha_day" db:"gotcha_day"`
+        Bio        *string     `json:"bio" db:"bio"`
+        Weight     *unit.Pound `json:"weight" db:"weight"`
+    }
+    
+    // Cats is a list of Cats
+    type Cats []Cat
+    ```
 
-// NewTables is not required by pop and may be deleted
-type NewTables []NewTable
+   Some things to note:
+    1. The field names here are pascal case, while in the struct tags, we use the snake case version. 
+    2. Notice how we used `string` for `Name`, while we used `*string` for `Bio`. We have a loose convention around
+       using pointers for optional (nullable) fields, and the concrete type for required fields.
+        1. One important thing to note here is that this will affect patch requests if you want the API caller to be 
+           able to make a request without passing in all the non-nullable fields each time. Some options which we use 
+           for existing models:
+            1. Make the field a pointer on the model. This means you will need to have validation somewhere though to
+               ensure that we save a valid value to the DB in the end.
+            2. Follow a pattern similar to the
+               [PPM shipment updater](https://github.com/transcom/mymove/blob/088b87605f19b18b022ff718fcbd5eb8f6962585/pkg/services/ppmshipment/ppm_shipment_updater.go#L44-L49)
+               which merges the new and old shipment _before_ validating the shipment.
+    3. We define the plural `type` under the main struct as a slice of the type.
```

Comment:
I like this convention, though I know not all our models follow this pattern. I wonder if we should note that or not. Maybe it's enough to say that it should be there.

Reply:
I didn't realize we had some without that. I think it's good to call out because I'm sure others will come across the different ones and wonder which to follow.

Reply:
Contradicting my self. I should clarify I like that we call out the pattern, I'm indifferent to the pattern itself. 🤦  sorry for the confusion